bdns/dns.go

```diff
@@ -244,8 +263,10 @@ func NewTest(
 	stats prometheus.Registerer,
 	clk clock.Clock,
 	maxTries int,
-	log blog.Logger) Client {
-	resolver := New(readTimeout, servers, stats, clk, maxTries, log)
+	log blog.Logger,
+	tlsConfig *tls.Config,
```

Comment:
It appears that every caller or NewTest passes nil for this new argument. It seems like either this argument should be omitted (and just pass nil to New directly), or a new test case should be added that exercises a custom tlsConfig.

Reply:
It's passed as non-nil in cmd/boulder-va/main:

https://github.com/letsencrypt/boulder/blob/302e2f0b1510a54cae130c453f1ac866f67f1edb/cmd/boulder-va/main.go#L110-L118

NewTest is used not just for unittests. It's also used for integration tests, to allow validating otherwise private addresses (like `10.*`).